src/x/xHilbertTransform.ts

```diff
@@ -1,26 +1,66 @@
+import { DoubleArray } from 'cheminfo-types';
+import FFT from 'fft.js';
+
+import { xCheck } from './xCheck';
+
 /**
  * Performs the Hilbert transform
  * @returns A new vector with 90 degree shift regarding the phase of the original function
  */
 
-export function xHilbertTransform(
-  input: number[],
+export function xHilbertTransform(array: DoubleArray) {
+  xCheck(array);
+  if (Math.log2(array.length) % 1 === 0) {
+    return hilbertTransformWithFFT(array);
+  } else {
+    return hilbertTransform(array);
```

Comment:
You didn't find a way to make a zero filling to next power of 2, make the hilbertTransformWithFFT and then cut the result ?
@jobo322 Could you give us your opinion about the feasability ?

Reply:
we have two functions not exposed in nmr-processing

```
function nextPowerOfTwo(n: number) {
  if (n === 0) return 1;
  n--;
  n |= n >> 1;
  n |= n >> 2;
  n |= n >> 4;
  n |= n >> 8;
  n |= n >> 16;
  return n + 1;
}
```

```
function isPowerOfTwo(n: number) {
  return n !== 0 && (n & (n - 1)) === 0;
}
```

we could move those functions to spectra-processing and use it in combination with the [xreimZeroFilling](https://github.com/mljs/spectra-processing/blob/main/src/xreim/xreimZeroFilling.ts) function

Reply:
xreimZeroFilling is redundant in this case